crates/recursor/src/recursor.rs

```diff
@@ -37,6 +38,7 @@ impl Default for RecursorBuilder {
         Self {
             ns_cache_size: 1024,
             record_cache_size: 1048576,
+            recursion_limit: 12,
```

Comment:
Would be good to comment on how we arrived at this value (if you made it up, some reasoning would be nice).